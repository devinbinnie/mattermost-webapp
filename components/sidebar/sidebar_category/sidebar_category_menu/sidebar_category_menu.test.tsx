// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';

import {CategoryTypes} from 'mattermost-redux/constants/channel_categories';

import {shallowWithIntl} from 'tests/helpers/intl-test-helper';

import SidebarCategoryMenu from './sidebar_category_menu';

describe('components/sidebar/sidebar_category/sidebar_category_menu', () => {
    const baseProps = {
        category: {
            id: 'category1',
            team_id: 'team1',
            type: CategoryTypes.CUSTOM,
            display_name: 'custom_category_1',
        },
        canCreatePublicChannel: true,
        canCreatePrivateChannel: true,
        isMuted: false,
        onToggle: jest.fn(),
        actions: {},
    };

    test('should match snapshot', () => {
        const wrapper = shallowWithIntl(
            <SidebarCategoryMenu {...baseProps}/>
        );

        expect(wrapper).toMatchSnapshot();
    });

    test('should match snapshot when category is favorites', () => {
        const props = {
            ...baseProps,
            category: {
                ...baseProps.category,
                type: CategoryTypes.FAVORITES,
            },
        };

        const wrapper = shallowWithIntl(
            <SidebarCategoryMenu {...props}/>
        );

        expect(wrapper).toMatchSnapshot();
    });

    test('should match snapshot when user does not have permissions', () => {
        const props = {
            ...baseProps,
            canCreatePrivateChannel: false,
            canCreatePublicChannel: false,
        };

        const wrapper = shallowWithIntl(
            <SidebarCategoryMenu {...props}/>
        );

        expect(wrapper).toMatchSnapshot();
    });
});
