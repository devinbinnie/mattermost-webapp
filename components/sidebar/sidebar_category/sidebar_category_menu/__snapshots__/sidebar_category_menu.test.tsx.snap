// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/sidebar/sidebar_category/sidebar_category_menu should match snapshot and contain correct buttons 1`] = `
<SidebarMenu
  ariaLabel="Category Menu"
  buttonAriaLabel="Category Menu"
  id="SidebarCategoryMenu-category1"
  onToggle={[MockFunction]}
  tooltipText="Category options"
>
  <MenuGroup>
    <MenuItemAction
      icon={
        <i
          className="icon-pencil-outline"
        />
      }
      id="rename-category1"
      onClick={[Function]}
      show={true}
      text="Rename Category"
    />
    <MenuItemAction
      icon={
        <i
          className="icon-plus"
        />
      }
      id="createChannel-category1"
      onClick={[Function]}
      show={true}
      text="Create Channel"
    />
  </MenuGroup>
  <MenuGroup>
    <MenuItemAction
      icon={
        <i
          className="icon-folder-plus-outline"
        />
      }
      id="create-category1"
      onClick={[Function]}
      show={true}
      text="Create Category"
    />
  </MenuGroup>
  <MenuGroup>
    <MenuItemAction
      icon={
        <i
          className="icon-bell-outline"
        />
      }
      id="delete-category1"
      isDangerous={true}
      onClick={[Function]}
      show={true}
      text="Delete Category"
    />
  </MenuGroup>
</SidebarMenu>
`;

exports[`components/sidebar/sidebar_category/sidebar_category_menu should match snapshot when category is favorites 1`] = `
<SidebarMenu
  ariaLabel="Category Menu"
  buttonAriaLabel="Category Menu"
  id="SidebarCategoryMenu-category1"
  onToggle={[MockFunction]}
  tooltipText="Category options"
>
  <MenuGroup />
  <MenuGroup>
    <MenuItemAction
      icon={
        <i
          className="icon-folder-plus-outline"
        />
      }
      id="create-category1"
      onClick={[Function]}
      show={true}
      text="Create Category"
    />
  </MenuGroup>
</SidebarMenu>
`;

exports[`components/sidebar/sidebar_category/sidebar_category_menu should match snapshot when user does not have permissions 1`] = `
<SidebarMenu
  ariaLabel="Category Menu"
  buttonAriaLabel="Category Menu"
  id="SidebarCategoryMenu-category1"
  onToggle={[MockFunction]}
  tooltipText="Category options"
>
  <MenuGroup>
    <MenuItemAction
      icon={
        <i
          className="icon-pencil-outline"
        />
      }
      id="rename-category1"
      onClick={[Function]}
      show={true}
      text="Rename Category"
    />
  </MenuGroup>
  <MenuGroup>
    <MenuItemAction
      icon={
        <i
          className="icon-folder-plus-outline"
        />
      }
      id="create-category1"
      onClick={[Function]}
      show={true}
      text="Create Category"
    />
  </MenuGroup>
  <MenuGroup>
    <MenuItemAction
      icon={
        <i
          className="icon-bell-outline"
        />
      }
      id="delete-category1"
      isDangerous={true}
      onClick={[Function]}
      show={true}
      text="Delete Category"
    />
  </MenuGroup>
</SidebarMenu>
`;
